integration-tests/narayana-jta/src/test/java/io/quarkus/narayana/jta/TransactionalTestCase.java

```diff
@@ -0,0 +1,22 @@
+package io.quarkus.narayana.jta;
+
+import static org.hamcrest.core.Is.is;
+
+import org.hamcrest.text.MatchesPattern;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.common.http.TestHTTPEndpoint;
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.RestAssured;
+
+@QuarkusTest
+@TestHTTPEndpoint(TransactionalResource.class)
+public class TransactionalTestCase {
+
+    @Test
+    public void test() {
+        RestAssured.when().get("/uid").then().assertThat().body(MatchesPattern.matchesPattern("[:0-9a-f]+"));
```

Comment:
Our Uid format is internal and if we change then this will break.
What advantage does testing uid format provide.
However the test that verifies that the `@Transactional` annotation correctly starts a transaction is sufficient,

Reply:
Yeah, I see. It is just used to verify the fix for https://github.com/quarkusio/quarkus/issues/10180. We can only see the warning messages when the ```SocketProcessId``` can not be loaded in the native mode, but the test will not fail. So I just use the ```[:0-9a-f]+``` to check there is not ```-1``` in the uid which means the process_id is correct. Do you have any better way to check the value of process_id ?

Reply:
The test method TransactionalResource.status() is marked with `@Transactional` so we will try to start a transaction.
Starting a transaction will then attempt to load the `SocketProcessId` class and if the class is not available then we throw an exception. This will cause the test that invokes `TransactionalResource.status()` to fail which again is what you want.

In other words there is no need for any more tests since the test `TransactionalTestCase#test()` is sufficient.

Note also that the calls `arjPropertyManager.getCoreEnvironmentBean().getProcessImplementation();`, `Utility.getpid()` and `new Uid()` would all do the same but *we do not need to test any of those* since starting a transaction is enough.

Reply:
Just as I said, it will not throw an exception if the ```SocketProcessId``` is not available and only show the warning mesage. So the test does not fail. The process_id will follow to ```-1``` and that is the reason I try to check the uid.

Reply:
see https://github.com/jbosstm/narayana/blob/master/ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/utils/Utility.java#L287

Reply:
I'm still not clear on why you want to test our process id class, it's private.
You should only be testing that you are able to start a new transaction.

If you really did need to check SocketProcessId (though I don't see any benefit in doing so, notice that we don't test any of the many other classes that we configure in the deployment processor, `NarayanaJtaProcessor`) then something like following would test it:

```java
        try {
            int pid = com.arjuna.ats.arjuna.utils.Utility.getpid();
            Assertions.assertNotEquals(-1, pid);
            Class clazz = arjPropertyManager.getCoreEnvironmentBean().getProcessImplementation().getClass();
            com.arjuna.ats.internal.arjuna.utils.SocketProcessId.class.getName();
            // success
        } catch (Exception e) {
            Assertions.fail("expectation failed", e);
        }
```

But like I say I would advise against testing it like this.

Reply:
OK, I undrestand.

Yeah, it should be better to remove this case for Uid in the integration tests since it is just the internal one in Narayana. Also note there is only WARN messages in the surefire tests output when the SocketProcessId can not be loaded in the native mode without this fix, the status test case will not fail.